feat(toolbar): Add setup instructions for the Dev Toolbar using the react hook useSentryToolbar (#12845)
getsentry/sentry-docs: docs/product/sentry-toolbar/faq.mdx

---
title: FAQ
sidebar_order: 30
description: "Frequently asked questions about the Sentry Toolbar."
---

<Expandable permalink title="In what environments should I enable the Toolbar?">

Since the Sentry Toolbar will be visible to users within your app, it's important to consider which environments should render it.

If your web application requires authentication to access:

> - In development and staging, always initialized the Sentry Toolbar.

Reviewer: ```suggestion
- In development and staging, always initialize the Sentry Toolbar.
```

> - In production conditionally initialize the Sentry Toolbar when an employee is logged in.

Reviewer: ```suggestion
- In production, conditionally initialize the Sentry Toolbar when an employee is logged in.
```

If you web application does not require authenticaion:

> - In development and staging environments initialize the Toolbar at all times.

Reviewer: ```suggestion
- In development and staging environments, initialize the Toolbar at all times.
```

- In production environments, do not initialize the Toolbar.

Initializing the Sentry Toolbar allows all developers and testers to quickly go from the page they're looking at, back to Sentry for further debugging.
In production it can make it easier for developers to reproduce issues, but it should not be initialized for all users of the site -- only when an employee/engineer/etc visits.

Once you decide where and when you want the Toolbar to appear, you'll write those conditions into your codebase. The specific implementation is something you'll need to write based on how your app works and how your team is set up.

</Expandable>

<Expandable permalink title="How can I conditionally initialize the Toolbar?">

> Implementing the specific conditions for initializing the Toolbar will vary from app to app and which ever framework or template library is in use.

Reviewer: ```suggestion
Implementing the specific conditions for initializing the Toolbar will vary from app to app and whichever framework or template library is in use.
```

> For example, the conditions to show show the Toolbar in development and staging might look like this, if written in JavaScript:

Reviewer: ```suggestion
For example, the conditions to show the Toolbar in development and staging might look like this, if written in JavaScript:
```
